### PR TITLE
TM-89 investigate the dbpedia api and create adapter

### PR DESCRIPTION
**Pull request by dan0bar**

### CODE REVIEW

File: be/src/resolvers/dbpedia-resolve.service.ts [hunk 1]
```diff
@@ -1,22 +1,129 @@
...
+      //     })
+      // }
+
+      // running = false
+      return Promise.resolve({data: response})
+    }
+    ).catch(e => 
+      // console.error(e);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      ServiceErrorFactory.requestTimeout(e.toString())
```
Comment (ManuelTS) [inline be/src/resolvers/dbpedia-resolve.service.ts:76]:
```suggestion
return ServiceErrorFactory.requestTimeout(e.toString());
```
to reject the Promise and communicate the error, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

File: be/src/resolvers/dbpedia-resolve.service.ts [hunk 2]
```diff
@@ -1,22 +1,129 @@
...
+    );
+
     // TODO TM-89, use DBpedia adapter here
```
Comment (ManuelTS) [inline be/src/resolvers/dbpedia-resolve.service.ts]: You have done TM-89, so this can be removed

**Review by ManuelTS — CHANGES_REQUESTED** [review]
Only minor stuff as said by @ja-fra in the daily, glad you have optimized the 3 for loops into a single one, good code!

File: be/src/resolvers/dbpedia-resolve.service.ts (hunk 1 above)
Comment (dan0bar) [reply be/src/resolvers/dbpedia-resolve.service.ts:76]: It doesn't let write return before the Error and insists on keeping like that.

Comment (ja-fra) [reply be/src/resolvers/dbpedia-resolve.service.ts:76]: There are two ways to write this:
```
.catch(e => ServiceErrorFactory.requestTimeout(e.toString()))
```
or
```
.catch(e => { return ServiceErrorFactory.requestTimeout(e.toString()); })
```

**Review by ja-fra — APPROVED** [review]

**Pull request merged**